core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java

```diff
@@ -107,4 +122,16 @@ public static boolean ancestorOf(Table table, long snapshotId, long ancestorSnap
 
     return newFiles;
   }
+
+  public static int existingDataFilesCount(Snapshot snapshot) {
+    int totalExistingDataFilesCount = 0;
+
+    if (snapshot != null) {
+      for (ManifestFile dataManifest : snapshot.dataManifests()) {
+        totalExistingDataFilesCount += dataManifest.existingFilesCount();
```

Comment:
The logic here should be:

```java
  totalExistingDataFilesCount += dataManifest.existingFilesCount();
  totalExistingDataFilesCount += dataManifest.snapshotId() != snapshot.snapshotId() ? dataManifest.addedFilesCount() : 0;
```

Reply:
as I moved implementing fullScan mode OUT of this PR - i marked this for future & linked up to the right work item.